src/components/PageSettingsModal/ResourcePageSettingsModal.tsx

```diff
@@ -0,0 +1,335 @@
+import {
+  Input,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  Box,
+  Text,
+  Center,
+  Spinner,
+  FormControl,
+  useDisclosure,
+} from "@chakra-ui/react"
+import { yupResolver } from "@hookform/resolvers/yup"
+import {
+  FormErrorMessage,
+  FormLabel,
+  ModalCloseButton,
+} from "@opengovsg/design-system-react"
+import axios from "axios"
+import { Breadcrumb } from "components/folders/Breadcrumb"
+import {
+  FormContext,
+  FormError,
+  FormTitle,
+  FormDescription,
+} from "components/Form"
+import FormFieldMedia from "components/FormFieldMedia"
+import { LoadingButton } from "components/LoadingButton"
+import MediaModal from "components/media/MediaModal"
+import ResourceFormFields from "components/ResourceFormFields"
+import { format } from "date-fns-tz"
+import _ from "lodash"
+import PropTypes from "prop-types"
+import { useEffect } from "react"
+import { useForm } from "react-hook-form"
+import slugify from "slugify"
+
+import elementStyles from "styles/isomer-cms/Elements.module.scss"
+
+import { getDefaultFrontMatter, pageFileNameToTitle } from "utils"
+
+import { PageSettingsSchema } from "./PageSettingsSchema"
+
+// axios settings
+axios.defaults.withCredentials = true
+
+interface ResourceModalParams {
+  siteName: string
+  fileName: string
+  resourceRoomName: string
+  resourceCategoryName: string
+}
+
+interface ResourcePageFrontMatter {
+  layout: string
+  title: string
+  permalink: string
+  date: string
+  // eslint-disable-next-line camelcase
```

Comment:
Hmm is it possible for us to use camelCase here? (unless its the cypress issue uh, then feel free to dismiss this comment)

Reply:
This is actually from the actual page front matter!